samples/greengrass/basic_discovery/main.cpp

```diff
@@ -373,5 +369,11 @@ int main(int argc, char *argv[])
         shutdownCompletedPromise.get_future().wait();
     }
 
+    /**
+     * Free the static default ClientBootstrap
+     * (will also free the static default HostResolver and EventLoopGroup)
+     */
+    Io::ClientBootstrap::ReleaseStaticDefault();
+
```

Comment:
I tried to find a way where this could be called automatically when the connection was closed/disconnected, similar to how it's done in this PR for Java, but I couldn't find a good spot to call it that wasn't in the C level CRT code, which I thought was a touch too low level since it's used across all SDKs and the change would only be needed for C++.